IObundle/iob-soc-tester
# Tester for SUTs and cores

This PR makes the system in this repository capable of Testing SUT's and Cores.

It also adds example firmware and configuration for the configurable version of the REGFILEIF peripheral (alongside the IOBNATIVEBRIDGEIF peripheral). This enables the SUT to have dedicated write and read registers, with sizes according to the configuration file `iob_regfileif_swreg.vh`. Instructions for this peripheral are available in the `README.md` file of its repository.

To run the SUT system by itself, run `make sim`
To run the Tester+SUT system, run `make tester-sim`
To use this system to test a core (core under test or CUT):
- add this repository as a submodule of the CUT's repository;
- create the tester.mk configuration file in the CUT's repository;
- create the makefile target in the CUT's repository to run the Tester:
  - `make -C $(TESTER_DIR) sim TESTING_CORE=1`
  - $(TESTER_DIR) is the location of this repository (should be located under the submodules directory of the CUT's repository).
- An example core under test is available in: https://github.com/IObundle/iob-cut
